instance/instance.go

```diff
@@ -260,6 +272,32 @@ func (i *Instance) Prefix() string {
 	return i.Domain + "/"
 }
 
+// RegisterPassword replace the instance registerToken by a password
+func (i *Instance) RegisterPassword(pass string, tok string) error {
+	if i.RegisterToken == "" {
+		return errors.New("Missing registerToken")
+	}
+	if i.RegisterToken != tok {
+		return errors.New("Invalid registerToken")
+	}
+
+	hash, err := scrypt.GenerateFromPassword([]byte(pass), scrypt.DefaultParams)
```

Comment:
We should have a random salt along with the password to generate the scrypt hash